docs/40_smart-contract-guides/92_read-only-queries.md

```diff
@@ -0,0 +1,337 @@
+
+Once the tutorial is completed you should be able to create and call a read-only query.
+
+## Before you begin
+This tutorial requires the following:
+
+* The EOSIO platform software, version 2.2. or greater, click on this link for instructions on [installing EOSIO binaries](https://developers.eos.io/manuals/eos/latest/install/install-prebuilt-binaries)
+* You understand how to build and deploy smart contracts
+
+## Run a Local Single-Node Blockchain
+Run a local nodeos instance which produces blocks, creating a local single-node testnet, using these [instructions](https://developers.eos.io/manuals/eos/latest/nodeos/usage/development-environment/local-single-node-testnet).
+ 
+
+## Build a Smart Contract with a Read-Only Query 
+
+Create a simple smart contract which contains three read-only actions: 
+- `get` - Will return a value 
+- `getComp` - Will return a complex value
+- `getMore` - Will return a vector of values
+
+Define the actions as read-only with the `eosio::read_only` attribute. Click on the follwing link for more information [ABI code generator attributes explained](https://developers.eos.io/manuals/eosio.cdt/latest/best-practices/abi/abi-code-generator-attributes-explained)   
+
+
+```cpp
+#include <eosio/eosio.hpp>
+class [[eosio::contract]] hello : public eosio::contract {
+  public:
+
+     struct retVal {
+         std::string name;
+         uint32_t gender;
+         uint32_t age;
+      };
+ 
+      using eosio::contract::contract;
+
+
+      [[eosio::action, eosio::read_only]] std::string get() {
+         return "Bob";
+      }
+
+
+      [[eosio::action, eosio::read_only]] retVal getComp() {
+         retVal rv;
+
+         rv.name = "Bob";
+         rv.gender = 1;
+         rv.age = 99;
+
+         return rv;
+      }
+
+
+      [[eosio::action, eosio::read_only]] std::vector<retVal> getMore() {         
+         std::vector<retVal> ret;
+         
+         retVal rv;
+         
+         rv.name = "Bob";
+         rv.gender = 1;
+         rv.age = 99;
+         ret.push_back(rv);
+
+         rv.name = "Alice";
+         rv.gender = 0;
+         rv.age = 19;
+         ret.push_back(rv);
+
+         return ret;
+      }
+};
+```
+
+Build the smart contract and deploy the smart contract to a blockchain using [how to compile a smart contract via the CLI](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/compile/compile-a-contract-via-cli) and [how to deploy a smart contract.](https://developers.eos.io/manuals/eos/latest/cleos/how-to-guides/how-to-deploy-a-smart-contract) 
```

Comment:
I recommend prescribing deploying the smart contract to the 'scholder' account so it's more likely the subsequent steps will work.  I would expect to see an explicit command in a tutorial as well, especially since the linked reference contains no samples.

Reply:
a prerequisite is "You understand how to build and deploy smart contracts" - I've updated this to include a link to the hello world tutorial which explains in detail building and deploying.

Have updated the Before you begin section to include:
running a blockchain
accounts and permissions
building and deploying smart contracts

and links for detailed info on all of these.